### PR TITLE
WIP: test_redirect.

### PR DESCRIPTION
**Pull request by RatanShreshtha**
Initially I was getting error `__init__() got an unexpected keyword argument 'redirect'` for `HTTPResponse` in `src/urllib3/_async/response.py` so I added a parameter `redirect=True i` in init for `HTTPResponse` and now when I am running it its giving me 303 when redirect is either True or False.

In file test/with_dummyserver/test_connectionpool.py
```
def test_redirect(self):
        r = self.pool.request('GET', '/redirect', fields={'target': '/'}, redirect=False)
        self.assertEqual(r.status, 303) # r.status is 303

        r = self.pool.request('GET', '/redirect', fields={'target': '/'})
        self.assertEqual(r.status, 200) # r.status is also 303 here
        self.assertEqual(r.data, b'Dummy server!')
```
Do we need to write the code for redirect also if so where are we supposed to write it ?

### CODE REVIEW

**Comment by pquentin** [comment]
When grepping for redirect, I see [code that handles redirects in PoolManager](https://github.com/python-trio/urllib3/blob/e85728ceb289dffd74d5e44b0c236ca9e9dbce9f/src/urllib3/_async/poolmanager.py#L328-L363). But the test here is about HTTPConnectionPool, which are managed by PoolManager, so that code won't kick in.

When I see surprising behavior like this, I run the same test from the master branch. That allowed me to see that [there is redirect handling in HTTPConnectionPool](https://github.com/urllib3/urllib3/blob/2a0957ea27e966166f81f88693af8e2f87d19fb6/src/urllib3/connectionpool.py#L678-L705). Somehow, that code got removed.

Was it removed in the v2 branch? Or in the bleach-spike branch that is based on v2? git log -p shows me that the redirect handling code was removed by https://github.com/urllib3/urllib3/pull/1146 in the v2 branch. It makes sense to only allow redirects in one place, and remove them from HTTPConnectionPool. In that PR, the test we're trying to fix here got removed: https://github.com/urllib3/urllib3/pull/1146/files

How did it get added back? It got added back by this [Merge from master](https://github.com/python-trio/urllib3/commit/e1d8b60cf96cb969cb01669bc8b29b3012995edb#diff-cf93a06bf593988c32c42644fa987a3bR395) commit. Okay, so the fix is to... remove this test! And looking at https://github.com/urllib3/urllib3/pull/1146 more, it looks like all the other changes were kept. So you only need to remove `test_redirect` here (and undo you response.py change).

**Comment by RatanShreshtha** [comment]
Thanks @pquentin for explaining, how to debug such cases so nicely :+1:

**Comment by pquentin** [comment]
Thanks! You still have an unrelated change in src/urllib3/_backends/sync_backend.py where you define two methods twice.

File: test/with_dummyserver/test_connectionpool.py [hunk 1]
```diff
@@ -329,6 +329,7 @@ def test_socket_options(self):
         conn = pool._new_conn()
         conn.connect()
         s = conn._sock
+
```
Comment (pquentin) [inline test/with_dummyserver/test_connectionpool.py]: You still have this extra line. I'll merge the changes without this line, but please remove it. :)

**Comment by pquentin** [comment]
Thanks! Merged manually in the bleach-spike branch.

**Pull request closed without merging**